### ISSUE TITLE
Add `ignoreDatesOnCi` option to `expiring-todo-comments` rule

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#1541]
Context: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1540

**Comment by fregante** [comment]
I'm not sure if it ever makes sense to _break_ on a certain date, whether locally or on CI.

I agree it makes sense to throw an error when a dependency has been updated for example, or when you're about to release a breaking version, but not just because _time has passed_, because that forces me to fix it immediately.

For example running `xo && ava` is impossible locally even with this exception, which makes it impossible to release a package with `npm test && npm version && npm publish`

I think `expiring-todo-comments` is nice on paper but in practice it has very limited utility. One of the conditions ("when the current package’s version changes") isn't even truthy until _after_ the publishing, which leads to awkward feature requests like:

- https://github.com/sindresorhus/np/issues/585

In short, I think dates should not be part of this rule, or at least be disabled by default (in "recommended")

**Comment by sindresorhus** [comment]
I agree, it's not optimal. I kinda think the rule should always be a warning (we should probably change the recommended config).

**Comment by fregante** [comment]
The problem with making that a warning is that it's not really different from regular TODO comments. Those already cause warnings, so what is the difference with this rule?

That would only work if ESLint's native rule excluded these expiring comments. This way, Unicorn would then show the warnings _only after they expire._

Today:

```
Warning: found comment: TODO: fix this line
```


Tomorrow:

```
Warning: found comment: TODO: fix this line
Warning: found expired comment: TODO[2023-02-10]: we can remove this line
```

I think this is the only setup where expiring date-based comments would make sense: _hidden_ until the date.

**Comment by sindresorhus** [comment]
Fixed by https://github.com/sindresorhus/eslint-plugin-unicorn/commit/5e31bba71b597a4078c346388335da4e9550e399

**Comment by fregante** [comment]
I still think that the rule should be reduced to just `constrained-comments` and remove all support for dates and non-constrained comments. 

- no more `warning`s on any TODO
- date-based TODO are just troublesome

Those are the ones that pull the rule into `warning` territory and break the core **"constraint"** part of the remaining conditions that should be `error` as by definition.

By intent:

```js
// No TODOs allowed
"no-warning-comments": "error"

// TODO reminders
"no-warning-comments": "warn",

// TODO reminders + later requirements
"no-warning-comments": "warn",
"unicorn/constrained-comments": "error"

// Requirements only (XO default)
"unicorn/constrained-comments": "error"
```

Dates have no place in this, they either break CI at a random date or just add to the list of warnings, which don't really have to be date-kept imho
